sync/src/main/java/tech/pegasys/teku/sync/PeerSyncBlockRequest.java

```diff
@@ -37,7 +37,15 @@ public PeerSyncBlockRequest(
 
   @Override
   public SafeFuture<?> onResponse(final SignedBeaconBlock response) {
-    slotOfLastBlock = Optional.of(response.getSlot());
+    UnsignedLong newBlockSlot = response.getSlot();
+    if (maybeSlotOfLastBlock.isPresent()) {
+      UnsignedLong slotOfLastBlock = maybeSlotOfLastBlock.get();
+      if (newBlockSlot.compareTo(slotOfLastBlock) <= 0) {
+        throw new OutOfOrderException(slotOfLastBlock, newBlockSlot);
+      }
+    }
```

Comment:
There are some other validations we should probably do as well:
* Check that the block slot is within the range we requested
* Check that the block slot matches the step we requested.  Something like: `(blockSlot - startSlot) % step == 0`

Reply:
Done.